readthedocs/doc_builder/backends/sphinx.py

```diff
@@ -390,64 +373,68 @@ def __init__(self, *args, **kwargs):
     sphinx_builder = 'readthedocssinglehtmllocalmedia'
-    sphinx_build_dir = '_build/localmedia'
+    sphinx_build_dir = "_readthedocs/htmlzip"
 
-    @restoring_chdir
-    def move(self, **__):
-        if not os.path.exists(self.old_artifact_path):
-            log.warning('Not moving localmedia because the build dir is unknown.')
-            return
-
-        log.debug('Creating zip file from path.', path=self.old_artifact_path)
+    def _post_build(self):
+        """Internal post build to create the ZIP file from the HTML output."""
+        sphinx_build_dir = os.path.join(self.project_path, self.sphinx_build_dir)
+        temp_zip_file = f"/tmp/{self.project.slug}-{self.version.slug}.zip"
         target_file = os.path.join(
-            self.target,
-            '{}.zip'.format(self.project.slug),
+            sphinx_build_dir,
+            f"{self.project.slug}.zip",
         )
-        if not os.path.exists(self.target):
-            os.makedirs(self.target)
-        if os.path.exists(target_file):
-            os.remove(target_file)
-
-        # Create a <slug>.zip file
-        os.chdir(self.old_artifact_path)
-        archive = zipfile.ZipFile(target_file, 'w')
-        for root, __, files in os.walk('.'):
-            for fname in files:
-                to_write = os.path.join(root, fname)
-                archive.write(
-                    filename=to_write,
-                    arcname=os.path.join(
-                        '{}-{}'.format(self.project.slug, self.version.slug),
-                        to_write,
-                    ),
-                )
-        archive.close()
+
+        # FIXME: ``arcname`` is invalid with this command still, but it's close
+        self.run(
+            "zip",
+            "-r",
+            temp_zip_file,
+            self.sphinx_build_dir,
+            cwd=self.project_path,
+            record=False,
+        )
+        self.run("rm", "-r", self.sphinx_build_dir, cwd=self.project_path, record=False)
+        self.run(
+            "mkdir", "-p", self.sphinx_build_dir, cwd=self.project_path, record=False
+        )
+        self.run("mv", temp_zip_file, target_file, cwd=self.project_path, record=False)
 
 
 class EpubBuilder(BaseSphinx):
 
-    type = 'sphinx_epub'
-    sphinx_builder = 'epub'
-    sphinx_build_dir = '_build/epub'
-
-    def move(self, **__):
-        from_globs = glob(os.path.join(self.old_artifact_path, '*.epub'))
-        if not os.path.exists(self.target):
-            os.makedirs(self.target)
-        if from_globs:
-            from_file = from_globs[0]
-            to_file = os.path.join(
-                self.target,
-                '{}.epub'.format(self.project.slug),
+    sphinx_builder = "epub"
+    sphinx_build_dir = "_readthedocs/epub"
+
+    def _post_build(self):
+        """Internal post build to cleanup EPUB output directory and leave only one .epub file."""
+        sphinx_build_dir = os.path.join(self.project_path, self.sphinx_build_dir)
```

Comment:
Seems odd to overwrite this class-level variable with  something the same name. Should the class-level one be more explicit that it's relative? 

```suggestion
        sphinx_build_dir = os.path.join(self.project_path, self.relative_build_dir)
```

Also seems odd it's named `sphinx` when nothing about it is sphinx specific?

Reply:
I'm using `relative_output_dir` and `absolute_output_dir` now. I think it's more explicit and will avoid confusions.